### PR TITLE
[v1.12] Author Backport of 29603 (examples: update guestbook example & test with new image registry)

### PR DESCRIPTION
**Pull request by mhofstetter**
The GCP Kubernetes Engine Samples migrated their image registry from Google Container Registry to Google Artifact Registry. Hence, the image gb-frontend from the guestbook example is no longer available.

Therefore, this commit changes the example to use the new registry. In addition it adopts the "leader/follower" naming.

Issue: GoogleCloudPlatform/kubernetes-engine-samples#209
Guestbook PR: GoogleCloudPlatform/kubernetes-engine-samples#194

PR from `main`: https://github.com/cilium/cilium/pull/29603

### CODE REVIEW

**Comment by mhofstetter** [comment]
/test-backport-1.12

File: examples/misc/guestbook/5-guestbook-controller.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
             "spec":{
                 "containers":[{
                     "name":"guestbook",
-                    "image":"gcr.io/google-samples/gb-frontend:v6",
+                    "image":"us-docker.pkg.dev/google-samples/containers/gke/gb-frontend:v5",
```
Comment (ghost) [inline examples/misc/guestbook/5-guestbook-controller.json:24]: Same comment as in https://github.com/cilium/cilium/pull/29603#discussion_r1414021621

Apart from changing the registry, the image is downgraded from v6 to v5. I assume it wasn't intentional?

Comment (mhofstetter) [reply examples/misc/guestbook/5-guestbook-controller.json:24]: `test/k8s/manifests/guestbook_deployment.yaml` (where i changed to image too) is used by the k8s-test "net_policies" (https://github.com/cilium/cilium/blob/v1.12/test/k8s/net_policies.go#L1924-L1931). If all tests succeed we should be good to continue with the `v5` image (which is the latest on that newer registry)

**Comment by mhofstetter** [comment]
/test-backport-1.12

**Comment by mhofstetter** [comment]
removed trailing character that let the tests fail. 

`K8sPolicyTest GuestBook Examples checks policy example` -> `FAIL: Error parsing JSON response: <br />`

**Review by ti-mo — APPROVED** [review]

**Comment by mhofstetter** [comment]
> @mhofstetter Is there an upstream commit to reference here?

@ti-mo I referenced the PR that will add the changes to `main` in to the PR description

Manual backport to v1.12 as this is the only branch that uses this image in an automated test (beside the example).

> PR from main: https://github.com/cilium/cilium/pull/29603

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
Every test failed :scream: 
```
/home/jenkins/workspace/Cilium-PR-K8s-1.16-kernel-4.9/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:527
Error parsing JSON response: <br />
<b>Fatal error</b>:  Uncaught Predis\Connection\ConnectionException: php_network_getaddresses: getaddrinfo failed: Name or service not known [tcp://redis-leader:6379] in /var/www/html/vendor/predis/predis/src/Connection/AbstractConnection.php:132
Stack trace:
#0 /var/www/html/vendor/predis/predis/src/Connection/StreamConnection.php(109): Predis\Connection\AbstractConnection-&gt;onConnectionError('php_network_get...', 0)
#1 /var/www/html/vendor/predis/predis/src/Connection/StreamConnection.php(154): Predis\Connection\StreamConnection-&gt;createStreamSocket(Object(Predis\Connection\Parameters), 'tcp://redis-lea...', 4)
#2 /var/www/html/vendor/predis/predis/src/Connection/StreamConnection.php(80): Predis\Connection\StreamConnection-&gt;tcpStreamInitializer(Object(Predis\Connection\Parameters))
#3 /var/www/html/vendor/predis/predis/src/Connection/AbstractConnection.php(81): Predis\Connection\StreamConnection-&gt;createResource()
#4 /var/www/html/vendor/predis/predis/src/Connection/StreamConnection.php(230): Predis\Connection\AbstractConnection-&gt; in <b>/var/www/html/vendor/predis/predis/src/Connection/AbstractConnection.php</b> on line <b>132</b><br />

Expected
    <*json.SyntaxError | 0xc000906138>: {
        msg: "invalid character '<' looking for beginning of value",
        Offset: 1,
    }
to be nil
/home/jenkins/workspace/Cilium-PR-K8s-1.16-kernel-4.9/src/github.com/cilium/cilium/test/k8s/net_policies.go:2034
```

**Comment by mhofstetter** [comment]
> Every test failed 😱

:fearful: yep, seems like the new image uses `redis-leader` instead of `redis-master`. Changed the deployments (example & test) to use the `leader/follower` naming.

Let's test again. :crossed_fingers:

**Comment by mhofstetter** [comment]
/test-backport-1.12

**Comment by mhofstetter** [comment]
/test-1.18-4.9

**Pull request merged**